code/modules/mob/living/silicon/say.dm

```diff
@@ -15,10 +15,19 @@
 	for(var/mob/M in GLOB.player_list)
 		if(M.binarycheck())
 			if(isAI(M))
-				var/renderedAI = "<span class='srt_radio binarysay'>Robotic Talk, <a href='?src=[REF(M)];track=[html_encode(name)]'><span class='name'>[name] ([desig])</span></a> <span class='message'>[message_a]</span></span>"
+				var/source = FALSE
+				if(M == src)
+					source = TRUE
+				var/renderedAI = "<span class='srt_radio binarysay[source ? " binarysayloud" : ""]'>Robotic Talk, <a href='?src=[REF(M)];track=[html_encode(name)]'><span class='name'>[name] ([desig])</span></a> <span class='message'>[message_a]</span></span>" //AI hears itself in loud mode.
 				to_chat(M, renderedAI)
+			else if(iscyborg(M))
+				var/mob/living/silicon/robot/borg = M
+				var/source = FALSE
+				if(src == borg.connected_ai)
+					source = TRUE
+				to_chat(M, "<span class='srt_radio binarysay[source ? " binarysayloud" : ""]'>[rendered]</span>") //Cyborg only hears master AI on loud mode.
 			else
-				to_chat(M, "<span class='srt_radio binarysay'>[rendered]</span>")
+				to_chat(M, "<span class='srt_radio binarysay[isAI(src) ? " binarysayloud" : ""]'>[rendered]</span>") //Human with binary channel hears all AIs on loud mode.
```

Comment:
This logic makes all AI chats to be loud to all borgs no matter if they're slaved or not.
You need a variable like you did `var/source = FALSE`, and make a condition if they're their master.

Reply:
I don't think I am interpreting the code wrong.
Again, the code should check if a message making AI is **their** master, not any other AI. It only makes all AI chats to be loud.

Reply:
oh, nevermind. I was wrong.